client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -34,8 +37,8 @@
     @Getter
     private final long automaticCheckpointIntervalMillis;
 
-    private final Map<Stream, StreamCut> startingStreamCuts;
-    private final Map<Stream, StreamCut> endingStreamCuts;
+    private final Map<String, StreamCut> startingStreamCuts;
```

Comment:
This still has the same problem as before. 
If they aren't passing the scope and you are inferring it, it should do so and then instantiate the `Stream` object. We shouldn't have an object whos type/content is ambiguous.